### DISCUSSION TITLE
Account creation with a multi-step form

### DISCUSSION THREAD

**Discussion by nephaest** [janko/rodauth-rails#210] [Q&A] [answered]
Hi fellow Rubyists and Janko,

I am starting a new app and would love to take this opportunity to introduce rodauth-rails as the authentication system instead of the Devise I am used to using.
I need to implement an onboarding process via a multi-step form (in fine feeding 2 different models).
In the first place I thought the approach could be to override the "registrations_controller" new and create actions but, not knowing much about the internals here, I can't figure out how to handle it and come to think this may not the appropriate approach.
How would you proceed to implement such a feature?

For some context, say we have the following:
- the User model with first_name, last_name and email attributes
- the UserProfile model with electric_meter_id, can_meter_id, etc.
- the SignupForm with all of the former attributes
Additionally, this process shall be compliant with email confirmation.

Thank you very much for any help one could provide

**Comment by janko** [comment, marked as answer]
I recommend just making this form multi-step on the client-side, showing and hiding different parts of the form, and then submitting all the data to the create account endpoint in the end.

The only use case I can imagine for wanting to make a multi-step form on the server side is remembering data that was already entered in case the user aborts. But that doesn't seem necessary for your use case.

**Reply by nephaest** [reply]
Hi Janko,
Thank you for your answer. I realize I may have oversimplified the context for the purpose of this discussion but I was actually expecting directions on how to handle the state server-side (so that it can be resumed at any time).
Would you know of a way to do so, I'd be happy if you could share your thoughts.
Thank you again!

EDIT:  I have a much clearer idea of how I want the onboarding to unfold and found an easy way to proceed. Thanks anyway!
